fenix/app/src/test/java/org/mozilla/fenix/browser/DownloadDialogUtilsTest.kt

```diff
@@ -0,0 +1,297 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.browser
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.spyk
+import io.mockk.verify
+import mozilla.components.browser.state.state.content.DownloadState
+import mozilla.components.browser.state.state.content.DownloadState.Status.COMPLETED
+import mozilla.components.browser.state.state.content.DownloadState.Status.FAILED
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class DownloadDialogUtilsTest {
+
+    @Test
+    fun `GIVEN final status and openInApp WHEN calling handleOnDownloadFinished and try to open the file fails THEN show an error message`() {
+        val downloadUtils = spyk(DownloadDialogUtils)
```

Comment:
Can we avoid using a spy? It is very easy to misuse them and if we are relying on them then I believe we need to spend more time thinking about how we structure our code to be testable. For example, finding a way to delegate out the `FenixSnackbar` code in a test.

In fact, on lines 47 to 53, we should not be using the spy and are no longer executing the real code path needed to catch real regressions.

```
        every {
            downloadUtils.shouldShowCompletedDownloadDialog(
                any(),
                any(),
                any(),
            )
        } returns true
```

Reply:
Thanks for bringing this up I removed the spy.